src/platform/datapath_raw_xdp.c

```diff
@@ -66,26 +67,41 @@ typedef struct XDP_INTERFACE {
     uint8_t RuleCount;
     CXPLAT_LOCK RuleLock;
     XDP_RULE* Rules;
-    XDP_QUEUE* Queues;
+    XDP_QUEUE* Queues; // An array of queues.
 } XDP_INTERFACE;
 
-typedef struct XDP_DATAPATH {
-    CXPLAT_DATAPATH;
-
-    BOOLEAN Running;
+typedef struct QUIC_CACHEALIGN XDP_WORKER {
+    const struct XDP_DATAPATH* Xdp;
     HANDLE CompletionEvent;
+    XDP_QUEUE* Queues; // A linked list of queues, accessed by Next.
+    uint16_t Index;
+} XDP_WORKER;
+
+void XdpWorkerAddQueue(_In_ XDP_WORKER* Worker, _In_ XDP_QUEUE* Queue) {
+    XDP_QUEUE** Tail = &Worker->Queues;
+    while (*Tail != NULL) {
+        Tail = &(*Tail)->Next;
+    }
+    *Tail = Queue;
+    Queue->Next = NULL;
+}
```

Comment:
Is there a specific reason for not using `CXPLAT_LIST_ENTRY`?

Reply:
I don't need a doubly-linked list.